### PR TITLE
Initial work on unstable-trait feature for stm32

### PR DESCRIPTION
**Pull request by lulf**
Implements async traits for exti for now.

### CODE REVIEW

**Comment by Dirbaio** [comment]
:rocket: !!!!

It seems you accidentally commited some stm32-metapac changes though

File: embassy-stm32/Cargo.toml [hunk 1]
```diff
@@ -51,11 +54,14 @@ time-driver-tim4 = ["_time-driver"]
 time-driver-tim5 = ["_time-driver"]
 
 # Reexport stm32-metapac at `embassy_stm32::pac`.
-# This is unstable because semver-minor (non-breaking) releases of embassy-stm32 may major-bump (breaking) the stm32-metapac version.
+# This is unstable because semver-minor (non-breaking) releases of embassy-stm31 may major-bump (breaking) the stm32-metapac version.
```
Comment (Dirbaio) [inline embassy-stm32/Cargo.toml:57]: stm31? :D

Comment (lulf) [reply embassy-stm32/Cargo.toml:57]: Omg my clumsy hands 😂

**Review by Dirbaio — APPROVED** [review]
LGTM! Thanks a lot for working on this!

- You accidentally commited stm32-metapac changes again
- Could you add building with+without unstable-traits to ci.sh? Like nrf, just build some with and some without, so we don't add even more lines...

**Pull request merged**